src/ucp/core/ucp_request.c

```diff
@@ -395,22 +395,19 @@ static ucp_md_map_t ucp_request_get_invalidation_map(ucp_ep_h ep)
     return inv_map;
 }
 
-int ucp_request_memh_invalidate(ucp_request_t *req, ucs_status_t status)
+static UCS_F_ALWAYS_INLINE ucp_mem_h
+ucp_request_get_memh(ucp_request_t *req, int extract, int is_fragment)
 {
-    ucp_ep_h ep                      = req->send.ep;
-    ucp_err_handling_mode_t err_mode = ucp_ep_config(ep)->key.err_mode;
-    ucp_worker_h worker              = ep->worker;
-    ucp_context_h context            = worker->context;
+    ucp_context_h context = req->send.ep->worker->context;
     ucp_mem_h *memh_p;
-    ucp_md_map_t invalidate_map;
-
-    if ((err_mode != UCP_ERR_HANDLING_MODE_PEER) ||
-        !(req->flags & UCP_REQUEST_FLAG_RKEY_INUSE)) {
-        return 0;
-    }
+    ucp_mem_h memh;
 
     /* Get the contig memh from the request basing on the proto version */
     if (context->config.ext.proto_enable) {
+        if (is_fragment) {
+            return req->send.rndv.mdesc->memh;
```

Comment:
`extract == true` case looks unhandled

maybe move it to new function and separate `is_fragment` earlier like in `ucp_request_memh_invalidate`?

Reply:
and it means that better to have separate function for this

Reply:
After our discussion I just removed `is_fragment` parameter from this function